specs/agents/tracing-instrumentation-azure-functions.md

```diff
@@ -0,0 +1,110 @@
+# Tracing Azure Functions
+
+An Azure Function Application can implement one or more handler functions that are executed via different trigger mechanisms (e.g. HTTP).
+Depending on these trigger mechanisms and the actual [Azure Functions runtime language and version](https://learn.microsoft.com/en-us/azure/azure-functions/supported-languages),
+the signature of these handler functions will vary, of course.
+
+In general however, there is always a *generic context* object available (e.g. [`FunctionContext`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.functions.worker.functioncontext)) that provides metadata and context for an instrumentation.
+
+Depending on the actual kind of invocation (trigger type) there might be an additional object to retrieve metadata and context from.
+In case of an HTTP invocation/trigger there is e.g. [`HttpRequestData`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.identitymodel.protocols.httprequestdata).
+
+## Generic Instrumentation
+
+In general, to instrument an Azure Functions application, we create transactions that wrap the execution of its handler methods. In cases where we cannot assume any trigger type or extract trigger-specific data (e.g. if the trigger type is unsupported),
+we wrap the handler method with a transaction, while using the generic context object to derive necessary fields.
+
+**Note:** This table represents generic values that agents SHOULD always provide regardless of the
+actual `faas.trigger.type`. For trigger types that are specifically supported (e.g HTTP) these values
+can vary on trigger-specific specifications apply.
+
+| Field | Value | Description | Source |
+| - | - | - | - |
+| `name` | e.g. `MySampleTrigger` | The transaction name. Use function name if trigger type is `other`. | *generic context* |
+| `type` | e.g. `request`, `messaging` | The transaction type. | Use `request` if trigger type is undefined. |
+| `outcome` | `success`/`failure` | Set to `failure` if a function error can be detected otherwise `success`. | |
+| `result` | `success`/`failure` |Set to `failure` if a function error can be detected, otherwise `success`. | Trigger specific. |
+| `faas.name` | e.g. `MySampleTrigger` | The function name. | *generic context* |
```

Comment:
[OTel semconv](https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/faas/#faas-resource-attributes) suggests `faas.name` should be `${functionAppName}/${functionName}`. What do you think about following that?  The app name is in the `WEBSITE_SITE_NAME` envvar. At least in JS, the function name is in `context.executionContext.functionName`, where `context` is the first arg to the handler function.

Reply:
Yes, good suggestion. Since we can have multiple function entry points in a function app, this is definitely more descriptive.
PR updated 👍 .